### ISSUE TITLE
[$250] [Due for payment 2025-04-18] MEDIUM: [Tracking] Implement more E2E tests and add new metrics

### ISSUE THREAD

**Issue by mountiny** [Expensify/App#30265]
The E2E tests are now set up and working again, lets expand on their options and add more flows and metrics:

- [ ] chat switching/opening for the first time
- [ ] typing
- [ ] scrolling chat messages
- [ ] number of renders when we open chat (it should be 1-2)
- [ ] searching
- [ ] signing in
- [ ] scrolling list of chats
- [ ] onyx tests

@Szymon20000 @hannojg 

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021913150357242377274</li>
        <li>Upwork Job ID: 1913150357242377274</li>
        <li>Last Price Increase: 2025-04-18</li>
    </ul>
</details>

**Comment by hannojg** [comment]
:wave:

**Comment by hannojg** [comment]
@perunt

**Comment by perunt** [comment]
👋

**Comment by mountiny** [comment]
Also Assigning @AndrewGable as he will most likely help us too 🤝 feel free to unassign if you prefer

**Comment by hannojg** [comment]
Update:

There was again a false positive on the e2e test runs. Added this PR trying to minimise those errors:

- https://github.com/Expensify/App/pull/30484

**Comment by hannojg** [comment]
Update:

First PR for adding tests for typing is up:

- https://github.com/Expensify/App/pull/30502

**Comment by mountiny** [comment]
Lies

**Comment by dangrous** [comment]
adding myself here for reviewing the chat switching one!

**Comment by hannojg** [comment]
Chat switching is merged 🎊 !

@perunt I think we can add "number of renders when we open chat (it should be 1-2)" to the same test as second metric?

Also next up would be the search page test!

**Comment by mountiny** [comment]
I think we might want to take a better look into how we are using the E2E tests now and make sure we already getting some use of it before adding more scenarios. In theory it can be very powerful tool, but so far we have not really been using the results of the tests to stop any PRs/ deploys. Come more convo [here](https://expensify.slack.com/archives/C03TQ48KC/p1703786837284899) cc @blimpich @roryabraham @iwiznia

**Comment by blimpich** [comment]
My 2 cents: E2E tests should be used incredibly sparingly. E2E tests are powerful but they're also expensive to make, maintain, and debug. I don't have much context on Expensify's E2E suite but I'd strongly lean in the direction of investing in many small, inexpensive unit tests instead of more E2E tests.

I'd also agree that if we aren't even using our current E2E test suite to stop deploys or cause code freezes, that there is no real reason why we even should care about the E2E test suite. The test suite as it stands is providing little to no value, since we currently can go a week with it being broken in `main` and no one noticing (see [this slack thread](https://expensify.slack.com/archives/C03TQ48KC/p1703786837284899)).

**Comment by mountiny** [comment]
This is not overdue, we are continuing as normal

**Comment by hannojg** [comment]
This is currently on hold by the effort to change how we mock the API. Discussion is here:

https://expensify.slack.com/archives/C035J5C9FAP/p1703237225747209

**Comment by mountiny** [comment]
Solution for the API mocking was provided, we need to test it

**Comment by mountiny** [comment]
@hannojg whats the latest here?

**Comment by hannojg** [comment]
- [We merged the PR to remove the API mocks](https://github.com/Expensify/App/pull/34792).
- [We merged the PR to fix the broken e2e test build pipeline](https://github.com/Expensify/App/pull/34786) 
- [We are currently waiting for the changes to become active in a latest release](https://expensify.slack.com/archives/C035J5C9FAP/p1706206646685159), to confirm the changes are working
- **So, next week we should be good to finally implement the remaining e2e tests!**

**Comment by hannojg** [comment]
Update: the e2e tests seem to be broken right now and we are working on finding out why.
[Conversation is happening here.](https://expensify.slack.com/archives/C035J5C9FAP/p1706704142097939)

**Comment by hannojg** [comment]
The e2e tests became broken again, [mentioned and fixed here](https://expensify.slack.com/archives/C035J5C9FAP/p1707318183490499).

We are working on:

- [Improving the observability of the e2e tests in case they brake or a regression gets detected](https://github.com/Expensify/App/pull/36004)
- [Making it easier to debug the issue with the e2e tests](https://github.com/Expensify/App/pull/35985)
- [Making the e2e test runner a little more forgiving](https://github.com/Expensify/App/pull/36025)

Once thats all merged and stable we will work on the next e2e tests (hopefully 🔜 ™️ )

**Comment by AndrewGable** [comment]
@hannojg is working on this

**Comment by hannojg** [comment]
After this PR was merged:

- https://github.com/Expensify/App/pull/37501

we are monitoring stability again, then moving on to implementing more tests!

**Comment by mountiny** [comment]
@hannojg what is the latest on this one?

**Comment by hannojg** [comment]
The tests are semi stable, currently 15% of them fail during running. Still want to iron that out!

**Comment by mountiny** [comment]
@hannojg what are the next steps on this issue from your side?

**Comment by hannojg** [comment]
We are now actively working on implementing the tests that align with the core metrics:

https://expensify.slack.com/archives/C05LX9D6E07/p1711455442814489?thread_ts=1711010712.155509&cid=C05LX9D6E07

**Comment by mountiny** [comment]
@hannojg how is this looking?

**Comment by hannojg** [comment]
We have two PRs open waiting for final review / clarifications:

- https://github.com/Expensify/App/pull/39394
- https://github.com/Expensify/App/pull/39417

**Comment by mountiny** [comment]
@hannojg What's next now?

**Comment by hannojg** [comment]
Next is to allocate time for this task in our team and move forward! Nothing blocking as far as I can see. We'll pick this up again next week!

**Comment by roryabraham** [comment]
We were chatting in slack about pointing these E2E tests at staging and using them to help figure out our "high water point" at which the performance degrades critically with an increasing number of reports across various devices.

**Comment by hannojg** [comment]
@roryabraham I remember we discussed doing a one-off report for those "high water points" [here](https://expensify.slack.com/archives/C05LX9D6E07/p1716928269801219?thread_ts=1716922862.986549&cid=C05LX9D6E07)

Is this something we should look at?

cc @kirillzyusko

**Comment by mountiny** [comment]
I believe @rinej is working on that

**Comment by hannojg** [comment]
@kirillzyusko lets finish this one!

Which tests are currently missing? I see that we are still missing:

- number of renders when we open chat (it should be 1-2)
- scrolling list of chats
- scrolling chat messages
- Onyx tests

I think for the onyx tests we wanted some tests where we don't load the full app, but just run some performance tests with mock data in onyx to catch performance regressions in onyx.

**Comment by kirillzyusko** [comment]
@hannojg yes, you are right. I think for number of renders we only track amount of re-renders in Composer.

Also I think we'll need to revise the approach for tracking number of re-renders, because now we rely on the fact, that if parent was re-rendered, then children that are not wrapped in memo will be re-rendered. But with `react-compiler` it doesn't work anymore, because it'll memoize everything automatically 👀

**Comment by kirillzyusko** [comment]
> Also I think we'll need to revise the approach for tracking number of re-renders, because now we rely on the fact, that if parent was re-rendered, then children that are not wrapped in memo will be re-rendered. But with react-compiler it doesn't work anymore, because it'll memoize everything automatically 👀

Just as a proof of my words - below I attached a report (after `react-compiler` was merged). And as we can see (first of all it really works 😅) that amount of renders went down from 2 to 0. I assume we have the same amount of re-renders, but since we were relying on an approach described above -> we have modified numbers now, so eventually at some point of time we'll need to revisit that.

<hr/>

## Performance Comparison Report 📊
### Significant Changes To Duration
_There are no entries_
### Meaningless Changes To Duration
<details>
<summary>Show entries</summary>

| Name                                          | Duration    |
| --------------------------------------------- | ----------- |
| App start nativeLaunch                        | 46.421 ms   |
| App start nativeLaunchEnd_To_appCreationStart | 50.533 ms   |
| App start appCreationEnd_To_contentAppeared   | 590.857 ms  |
| App start appCreation                         | 57.070 ms   |
| App start contentAppeared_To_screenTTI        | 985.122 ms  |
| App start TTI                                 | 1727.189 ms |
| App start runJsBundle                         | 441.492 ms  |
| App start regularAppStart                     | 0.022 ms    |
| Load Search Options                           | 213.[10](https://github.com/Expensify/App/actions/runs/9660966578/job/26648014660#step:17:11)0 ms  |
| Open Chat Finder Page TTI                     | 1237.410 ms |
| Composer typing rerender count                | 0.000 ms    |
| Chat opening                                  | 166.850 ms  |
| Chat TTI                                      | 565.710 ms  |
| Comment linking                               | 480.[11](https://github.com/Expensify/App/actions/runs/9660966578/job/26648014660#step:17:12)0 ms  |
</details>

<details>
<summary>Show details</summary>

| Name                                          | Duration                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      
| --------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
| App start nativeLaunch                        | **Baseline**<br/>Mean: 46.474 ms<br/>Stdev: 4.519 ms (9.7%)<br/>Runs: 39 39 41 41 41 41 41 42 42 42 43 43 43 43 43 44 44 44 44 44 44 44 44 44 45 45 45 45 46 46 46 46 47 47 47 47 47 48 48 48 48 49 49 50 50 50 51 51 51 51 52 52 54 54 57 58 59<br/><br/>**Current**<br/>Mean: 46.421 ms<br/>Stdev: 5.231 ms (11.3%)<br/>Runs: 39 40 40 40 40 40 41 41 42 42 42 42 42 42 42 42 43 43 43 43 43 43 44 44 45 45 45 46 46 46 46 46 46 46 46 47 47 47 48 48 48 49 50 50 51 51 52 52 52 52 53 54 54 55 59 60 61                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    
| App start nativeLaunchEnd_To_appCreationStart | **Baseline**<br/>Mean: 49.867 ms<br/>Stdev: 8.401 ms (16.8%)<br/>Runs: 36 37 37 38 39 40 40 41 41 42 42 42 43 43 44 44 44 44 44 45 45 45 46 46 46 47 48 48 48 49 49 49 50 51 51 51 52 52 52 52 52 53 53 54 54 56 57 57 57 58 59 59 60 60 63 63 64 66 69 75<br/><br/>**Current**<br/>Mean: 50.533 ms<br/>Stdev: 8.401 ms (16.6%)<br/>Runs: 38 39 39 40 40 41 41 41 42 42 42 43 43 43 44 44 44 45 45 45 45 47 47 47 47 47 47 48 48 49 49 50 51 51 52 52 52 53 53 53 53 54 54 55 55 56 57 57 57 57 57 60 60 61 62 67 69 70 70 72                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 
| App start appCreationEnd_To_contentAppeared   | **Baseline**<br/>Mean: 586.533 ms<br/>Stdev: 43.644 ms (7.4%)<br/>Runs: 497 506 514 516 520 524 529 529 534 537 538 539 540 545 551 554 561 564 575 576 576 578 580 581 582 582 582 586 587 588 589 589 590 590 591 595 596 596 602 603 610 611 6[12](https://github.com/Expensify/App/actions/runs/9660966578/job/26648014660#step:17:13) 613 617 617 617 618 622 622 627 627 639 642 645 648 656 662 685 690<br/><br/>**Current**<br/>Mean: 590.857 ms<br/>Stdev: 31.269 ms (5.3%)<br/>Runs: 519 524 530 537 538 543 544 546 553 555 565 574 577 577 577 579 580 580 582 582 584 586 587 587 587 588 588 592 592 593 594 596 597 597 598 598 602 610 610 611 612 615 616 618 620 621 621 623 626 627 628 629 631 637 650 655                                                                                                                                                                                                                                                                                                                                                                       
| App start appCreation                         | **Baseline**<br/>Mean: 61.390 ms<br/>Stdev: 10.279 ms (16.7%)<br/>Runs: 46 48 49 50 51 51 51 51 51 52 52 52 53 53 53 54 54 54 54 54 54 55 55 55 55 56 58 59 59 60 60 60 61 61 62 62 62 62 63 64 65 66 68 69 69 69 71 71 72 72 72 74 74 75 75 80 85 87 92<br/><br/>**Current**<br/>Mean: 57.070 ms<br/>Stdev: 6.948 ms (12.2%)<br/>Runs: 44 45 46 47 48 48 48 50 50 50 51 51 51 51 51 52 52 54 54 54 55 55 55 55 56 56 56 57 57 57 58 58 58 59 59 59 59 59 59 59 60 60 61 61 62 62 62 63 63 64 66 69 69 71 71 72 74                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            
| App start contentAppeared_To_screenTTI        | **Baseline**<br/>Mean: 1065.342 ms<br/>Stdev: 76.461 ms (7.2%)<br/>Runs: 907.8019419999328 915.2966060000472 922.8915379999671 923.1415490000509 936.7570619999897 941.0060360000934 949.8935799999163 982.3591420000885 983.5623409999534 990.5694039999507 998.6525149999652 999.3633300000802 999.6344999999274 1006.9684609998949 1007.7174[13](https://github.com/Expensify/App/actions/runs/9660966578/job/26648014660#step:17:14)000064 1016.4667450000998 1020.0574990001041 1022.4106940000784 1023.4612030000426 1023.9040620001033 1033.2889250000007 1041.210217999993 1044.919617000036 1048.57[14](https://github.com/Expensify/App/actions/runs/9660966578/job/26648014660#step:17:15)539999608 1050.9262329998892 1058.0397880000528 1064.8353329999372 1064.9087469999213 1065.360864999937 1065.9842719999142 1073.4[15](https://github.com/Expensify/App/actions/runs/9660966578/job/26648014660#step:17:16)3340000194 1081.0659119999036 1081.4825420000125 1081.8841289998963 1082.1053369999863 1082.308960000053 1082.649822999956 1090.1506209999789 1090.231924999971 1090.3712689999957 1106.3483090000227 1107.3801849999 1114.1298309999984 1115.2328929998912 1115.4641899999697 11[16](https://github.com/Expensify/App/actions/runs/9660966578/job/26648014660#step:17:17).5955270000268 1124.0819810000248 1124.3340419998858 1128.1248780
| App start TTI                                 | **Baseline**<br/>Mean: 1811.642 ms<br/>Stdev: 110.432 ms (6.1%)<br/>Runs: 1570.1415490000509 1605.7570619999897 1610.891537999967 1612.8935799999163 1613.2966060000472 1631.8019419999328 1643.5623409999534 1648.3591420000885 1649.3633300000802 1650.0060360000934 1677.7[17](https://github.com/Expensify/App/actions/runs/9660966578/job/26648014660#step:17:18)413000064 1692.9040620001033 1702.6344999999274 1716.5694039999507 1759.4667450000998 1761.1053369999863 1764.0574990001041 1769.4825420000125 1771.0659119999036 1774.0397880000528 1778.4106940000784 1779.968460999895 1791.210217999993 [18](https://github.com/Expensify/App/actions/runs/9660966578/job/26648014660#step:17:19)00.5714539999608 1803.4612030000426 1808.360864999937 1823.649822999956 1824.6525149999652 1828.9087469999213 1830.3712689999957 1836.7707030000165 1840.9[19](https://github.com/Expensify/App/actions/runs/9660966578/job/26648014660#step:17:20)617000036 1843.8841289998963 1847.4153340000194 1848.8353329999372 1851.1506[20](https://github.com/Expensify/App/actions/runs/9660966578/job/26648014660#step:17:21)9999789 1854.308960000053 1856.9842719999142 1862.2328929998912 1871.2889250000007 1875.1248780000024 1887.4641899999697 1891.1298309999984 1892.0819810000248 1894.3340419998858 1894.9262329998892 1895.9072310000192 1899.3483090000
| App start runJsBundle                         | **Baseline**<br/>Mean: 440.915 ms<br/>Stdev: 31.[21](https://github.com/Expensify/App/actions/runs/9660966578/job/26648014660#step:17:22)1 ms (7.1%)<br/>Runs: 368 379 387 395 395 397 402 403 406 406 409 409 414 414 419 420 423 425 425 427 431 433 433 434 435 436 440 442 443 444 444 445 447 447 447 448 448 449 454 455 456 458 458 460 463 468 468 469 471 473 473 475 478 481 482 489 493 508 513<br/><br/>**Current**<br/>Mean: 441.492 ms<br/>Stdev: 28.374 ms (6.4%)<br/>Runs: 384 385 386 387 397 399 400 402 408 409 412 416 420 420 425 426 427 430 430 432 432 432 434 437 440 440 441 445 446 447 447 448 448 448 449 450 452 452 453 454 454 455 455 455 457 459 465 467 469 470 470 472 475 476 481 487 489 498 504                                                                                                                                                                                                                                                                                                                                                               
| App start regularAppStart                     | **Baseline**<br/>Mean: 0.023 ms<br/>Stdev: 0.002 ms (10.3%)<br/>Runs: 0.017089999979361892 0.017699999967589974 0.01818799995817244 0.01827000011689961 0.018554000183939934 0.018718000035732985 0.01977499993517995 0.019896999932825565 0.01989799994044006 0.02030499977990985 0.02087399992160499 0.02123999991454184 0.021321999840438366 0.02140299999155104 0.021403000[22](https://github.com/Expensify/App/actions/runs/9660966578/job/26648014660#step:17:23)4381685 0.021646999986842275 0.021727999905124307 0.02176999999210[23](https://github.com/Expensify/App/actions/runs/9660966578/job/26648014660#step:17:24)85 0.021849999902769923 0.021850999910384417 0.022013999987393618 0.02[24](https://github.com/Expensify/App/actions/runs/9660966578/job/26648014660#step:17:25)61000131443143 0.02254199981689453 0.022582999896[25](https://github.com/Expensify/App/actions/runs/9660966578/job/26648014660#step:17:26)8116 0.02258300012908876 0.02266400004737079 0.022704999893903732 0.022745999973[26](https://github.com/Expensify/App/actions/runs/9660966578/job/26648014660#step:17:27)7317 0.022786000045016408 0.022948999889194965 0.02294900012202561 0.02298999996855855 0.023194000124931335 0.023[27](https://github.com/Expensify/App/actions/runs/9660966578/job/26648014660#step:17:28)399980276823 0.023314999882131815 0.0235190000385046 0.023558999877423048 0.023600999964401126 0.023640999803319573 0.023680999875068665 0.023681999882683158 0.024252000031992793 0.024373000022023916 0.024495000019669
| Load Search Options                           | **Baseline**<br/>Mean: 215.810 ms<br/>Stdev: 15.900 ms (7.4%)<br/>Runs: 187.6241859998554 188.15079699992202 189.3496509999968 190.11763500003144 190.5600179999601 191.2069500000216 191.7418619999662 192.7706299999263 193.29142200015485 199.338419999927[28](https://github.com/Expensify/App/actions/runs/9660966578/job/26648014660#step:17:29) 199.9901129999198 201.370483000[29](https://github.com/Expensify/App/actions/runs/9660966578/job/26648014660#step:17:30)71 202.27295000012964 205.46675600018352 205.87142000021413 207.3771569998935 207.7867439999245 208.11596699990332 209.21459899982437 209.2482089996338 210.2288000001572 210.27315[30](https://github.com/Expensify/App/actions/runs/9660966578/job/26648014660#step:17:31)0004557 212.57914200006053 212.72977699991[31](https://github.com/Expensify/App/actions/runs/9660966578/job/26648014660#step:17:32)6 212.9648849996738 213.21862800000235 213.52095600031316 214.11641399981454 215.16345199989155 215.73836300009862 217.7987069999799 218.44311500014737 218.49833199987188 218.6028639995493 218.76013199985027 219.09814399993047 221.778[32](https://github.com/Expensify/App/actions/runs/9660966578/job/26648014660#step:17:33)000004128 224.21106000011787 225.00195300020278 225.20874000014737 225.68843599967659 226.3619389999658 227.2277020001784 227.28361000027508 227.28812699997798 227.8553059999831 227.86226400034502 228.46329799992964 230.8099
| Open Chat Finder Page TTI                     | **Baseline**<br/>Mean: 1183.998 ms<br/>Stdev: 57.718 ms (4.9%)<br/>Runs: 1049.187[33](https://github.com/Expensify/App/actions/runs/9660966578/job/26648014660#step:17:34)79999306 1070.6500249998644 1091.[34](https://github.com/Expensify/App/actions/runs/9660966578/job/26648014660#step:17:35)07799997367 1093.3906260000076 1100.8560799998231 1106.7984219999053 1111.3887950000[35](https://github.com/Expensify/App/actions/runs/9660966578/job/26648014660#step:17:36)3 1114.4760750001296 1125.1623539999127 1134.828614000231 1137.7478849999607 1138.5943610002287 1139.1834720000625 1139.372802999802 1139.5280770002864 1139.805136000039 1141.2639580001123 1146.276041999925 1146.4478360000066 1151.9972340003587 1152.890585000161 1155.1714679999277 1155.269328000024 1156.8361410000362 1157.2795820003375 1157.4157310002483 1158.2241620002314 1169.0699880002066 1174.953288000077 1181.7361249998212 1184.7868249998428 1186.482340999879 1191.9135340000503 1199.40730800014 1203.1869720001705 1214.8201910001226 1215.7211919999681 1218.0220139999874 1218.7476820000447 1220.501221000217 1225.2565520000644 1226.3272310001776 1228.6660569999367 1228.710694000125 1229.2508960003033 1229.7907730001025 1232.983196000103 1240.5893149999902 12
| Composer typing rerender count                | **Baseline**<br/>Mean: 2.000 ms<br/>Stdev: 0.000 ms (0.0%)<br/>Runs: 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2<br/><br/>**Current**<br/>Mean: 0.000 ms<br/>Stdev: 0.000 ms (NaN%)<br/>Runs: 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 
| Chat opening                                  | **Baseline**<br/>Mean: 144.959 ms<br/>Stdev: 6.599 ms (4.6%)<br/>Runs: 134.66532400017604 135.12951700016856 135.66825400013477 1[36](https://github.com/Expensify/App/actions/runs/9660966578/job/26648014660#step:17:37).48018399998546 137.52022300008684 138.20890300022438 138.2253020000644 139.0252680000849 139.7583010001108 139.76721200020984 139.94966599997133 140.440510999877 140.51761899981648 140.68684900039807 140.8257650001906 140.87076900014654 141.05513500003144 141.07238799985498 141.22603400005028 141.54764899984002 141.60290500009432 141.6551510002464 141.7930899998173 142.13582399999723 142.19893399998546 142.75113999983296 143.33548999996856 143.66524200001732 143.85380100039765 145.12675000028685 145.1649989997968 145.6087239999324 145.70882199984044 145.71130400011316 145.83064700011164 146.2416179999709 146.3520510001108 146.36348400032148 147.00[37](https://github.com/Expensify/App/actions/runs/9660966578/job/26648014660#step:17:38)0300002396 147.28881900012493 147.55061899963766 147.90820399997756 147.92561799986288 149.02136199967936 150.31978400005028 151.28027400001884 153.69934100005776 153.95800799969584 154
| Chat TTI                                      | **Baseline**<br/>Mean: 476.442 ms<br/>Stdev: 30.748 ms (6.5%)<br/>Runs: 434.2028810000047 437.42981000011787 440.26863600034267 441.4828290003352 441.9324139999226 446.1614179997705 446.4310710001737 446.7055259998888 447.65071700001135 448.716349999886 449.24047899991274 449.6853030002676 449.82076000003144 450.2500409996137 450.9113360000774 450.969034999609 451.24296100018546 452.19966599997133 452.81559300003573 454.2888600002043 454.53576700016856 456.12874399963766 457.02852400019765 457.63704500021413 458.65694200014696 459.0605069999583 462.0355630000122 463.3727220003493 466.6648770002648 466.9698080001399 467.61844900017604 472.4313559997827 475.7039799997583 479.0117190000601 479.1413169996813 479.52579799992964 481.0275070001371 481.9576820000075 482.4648040002212 483.312663000077 484.4969489998184 485.2662769998424 485.91410299995914 492.203572999686 495.68957499973476 500.75052900006995 501.087728000246 503.28108800016344 504.24967499962077 504.2
| Comment linking                               | **Baseline**<br/>Mean: 479.941 ms<br/>Stdev: 94.333 ms (19.7%)<br/>Runs: 329.9139000000432 343.6412349999882 345.00870699994266 353.8523350004107 356.29203299991786 365.00968400016427 368.8347169999033 369.1029059998691 369.19584200065583 370.2426760001108 377.52616299968213 378.4616290004924 [38](https://github.com/Expensify/App/actions/runs/9660966578/job/26648014660#step:17:39)1.8841559998691 387.89355499949306 391.8345539998263 394.7826339993626 401.44868999999017 415.91259799990803 419.47282000025734 419.6573490006849 420.69120300002396 433.2966719996184 437.5493980003521 441.165731000714 449.463175999932 454.1970620006323 455.46671599987894 463.6658939998597 469.843913000077 470.22635999973863 470.660442000255 477.4665130004287 481.33044399973005 483.81404699990526 492.84175700042397 503.6965330000967 507.8522140001878 508.1464029997587 510.19746899977326 512.2441819999367 522.2788909999654 533.435344000347 535.5654710000381 537.1874599996954 540.5981049994007 548.4115810003132 551.3695480003953 564.5918780001812 568.8046059999615 571.7102
</details>

**Comment by dangrous** [comment]
hi friends! just checking in - where are we on this one? let me know if I can help in any way!

**Comment by kirillzyusko** [comment]
@dangrous I think we are currently (still) trying to fix e2e tests. And only after that we can start to write new e2e tests 👀

**Comment by dangrous** [comment]
how are we looking?

**Comment by kirillzyusko** [comment]
@dangrous Still fixing e2e tests 😅 Recent RN 0.75 broke them 😔

**Comment by hannojg** [comment]
Okay, so the tests are stable again, and we are collecting additional metrics.
Now we'd like to add more e2e performance regression tests. What other flows do you think are important to cover?
I feel like we currently don't really cover any money request flows?

We will add other tests like chat scrolling etc as mentioned in the issue's description soon!

**Comment by dangrous** [comment]
How are things going here?

**Comment by hannojg** [comment]
cc @kirillzyusko is working on that (please assign him as well). We've overhauled the e2e pipeline to be more robust. We have to fix one failing test, but are otherwise ready to write some more e2e tests this week (finally!!)

**Comment by dangrous** [comment]
awesome!

**Comment by kirillzyusko** [comment]
Yes, I should get some time to work on new e2e tests this week!

**Comment by hannojg** [comment]
We are actively working on extending the e2e test suite. Current next PR we're working on is:

- https://github.com/Expensify/App/pull/52751

**Comment by perunt** [comment]
It's alive. There are two new tests on the way:
https://github.com/Expensify/App/pull/52751?notification_referrer_id=NT_kwDOAuV5O7QxMzQzNjcyMzY1MDo0ODU5MzIxMQ#discussion_r1948711248
https://github.com/Expensify/App/pull/54625

**Comment by dangrous** [comment]
yep, reopening. chill, melvin

**Comment by dangrous** [comment]
Sounds like we're almost there on the PRs for this

**Comment by dangrous** [comment]
One down, one to go (right?)
#52751 is merged / on staging!
#54625 is still in progress. Do we have a status update on this one?

**Comment by perunt** [comment]
There was some migration work on the chat screen, so I paused on [this PR](https://github.com/Expensify/App/pull/54625). I’ll pick it back up on Monday and finish it

**Comment by jliexpensify** [comment]
Payment Summary

- C+: @allgandalf $250 (to be paid via Upworks)

Please apply here: https://www.upwork.com/jobs/~021913150357242377274

**Comment by jliexpensify** [comment]
Hmm not seeing anything @allgandalf - are you Rohan S.?

**Comment by jliexpensify** [comment]
@allgandalf No, but since you haven't added your Upworks Name or Profile to your GH profile, it makes it hard to work out who you are. I hire many contributors, and I've asked you to add your Upworks name to your GH profile before and you haven't. In the .MD, we do ask contributors to add their details in their GH profile.

**Comment by jliexpensify** [comment]
Paid, job closed.
